src/components/TransactionItemRow/ReceiptPreview/index.tsx

```diff
@@ -0,0 +1,56 @@
+import React, {useEffect, useRef, useState} from 'react';
+import ReactDOM from 'react-dom';
+import Animated, {FadeIn, FadeOut} from 'react-native-reanimated';
+import Image from '@components/Image';
+import useThemeStyles from '@hooks/useThemeStyles';
+
+function ReceiptPreview({source, hovered}: {source: string; hovered: boolean}) {
+    const [shouldShow, setShouldShow] = useState(false);
+    const debounceTimeout = useRef<NodeJS.Timeout | null>(null);
+    const body = document.querySelector('body');
+    const styles = useThemeStyles();
+
+    useEffect(() => {
+        if (hovered) {
+            debounceTimeout.current = setTimeout(() => {
+                setShouldShow(true);
+            }, 333);
+        } else {
+            if (debounceTimeout.current) {
+                clearTimeout(debounceTimeout.current);
+                debounceTimeout.current = null;
+            }
```

Comment:
I think this clearing block can be removed.
If we `setShouldShow(false)` this will make the picture disappear which seems to work. And the final cleanup will be done by the return function.